### DISCUSSION TITLE
Build plugin vs javac plugin

### DISCUSSION THREAD

**Discussion by delanym** [revelc/formatter-maven-plugin#714] [General]
The [Errorprone](https://errorprone.info) project allows patching java files during compilation.
It seems that for java anyway it would be more efficient to format the code as part of the compilation.
It would probably solve a lot of other issues like the ability to order imports, or perform save actions, or even run Intellij inspections.
What is your take on the possibility of a javac plugin formatter rather than a stand-alone build plugin?

**Comment by ctubbsii** [comment]
@delanym That sounds like a different project entirely. If somebody wants to do that, go right ahead, but it's unrelated to this project.
